lib/timers.js

```diff
@@ -483,152 +577,3 @@ exports.clearImmediate = function(immediate) {
     process._needImmediateCallback = false;
   }
 };
-
-
-// Internal APIs that need timeouts should use timers._unrefActive instead of
-// timers.active as internal timeouts shouldn't hold the loop open
-
-var unrefList, unrefTimer;
-
-function _makeTimerTimeout(timer) {
-  var domain = timer.domain;
-  var msecs = timer._idleTimeout;
-
-  L.remove(timer);
-
-  // Timer has been unenrolled by another timer that fired at the same time,
```

Comment:
It was necessary at the time of 9724047268afb96ba5435d98636ea745740fd12f.

The check is made unnecessary now, as you explained, because we used to [queue all internal/unref timers that need to timeout](https://github.com/nodejs/node/pull/4007/files#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eL592). So if their callback would `unenroll` them, they would still be queued for timeout, and thus we needed to check their `_idleTimeout` property.

Now with this PR even internal/unref timers [call their `_onTimeout` callback as their list is processed](https://github.com/nodejs/node/pull/4007/files#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eR158), and so if one timer `unenroll` another, that other timer will be removed from the list and not processed. Thus its `_onTimeout` callback won't be called.

So it seems that the comment at https://github.com/nodejs/node/pull/4007/files#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eR130 about the `listOnTimout` loop timer being "too early" because of unenrolled timers is not accurate, and should be removed, because these unenrolled timers won't even be processed in this loop.